src/middleware.ts

```diff
@@ -449,24 +459,24 @@ export const persist =
       )
     }
 
-    const thenableSerialize = toThenable(serialize)
+    const thenableSerialize = toThenable(options.serialize)
 
     const setItem = (): Thenable<void> => {
-      const state = partialize({ ...get() })
+      const state = options.partialize({ ...get() })
 
-      if (whitelist) {
+      if (options.whitelist) {
         ;(Object.keys(state) as (keyof S)[]).forEach((key) => {
-          !whitelist.includes(key) && delete state[key]
+          !options.whitelist?.includes(key) && delete state[key]
```

Comment:
do we need `?`?

Reply:
For some reason, typescript think `options.whitelist` can be `undefined` within the forEach

Reply:
Okay, then let's leave it. Will be removed in #598 anyway.